test/unit/transaction-runner-test.coffee

```diff
@@ -327,20 +328,63 @@ describe 'TransactionRunner', ()->
           done()
 
     describe 'when a test has been manually set to skip in a hook', () ->
+      clonedTransaction = null
 
-      beforeEach () ->
+      beforeEach (done) ->
         sinon.stub configuration.emitter, 'emit'
+
+        clonedTransaction = clone(transaction)
+
         runner = new Runner(configuration)
 
+        addHooks runner, [clonedTransaction], (err) ->
+          done err if err
+
+          runner.hooks.beforeHooks =
+            'Group Machine > Machine > Delete Message > Bogus example name' : [
+              (transaction) ->
+                transaction.skip = true
```

Comment:
should be `data`? or just local variable with the same name as the clonedTransaction / transaction?

Right now it is a bit confusing.